### DISCUSSION TITLE
can you give me example router params

### DISCUSSION THREAD

**Discussion by bobwatcherx** [flet-dev/flet#536] [General]
i have problem , and stuck use route params
this code
```
def route_change(route):
		page.views.clear()
		page.views.append(
			View(
				"/",
				[
				AppBar(title=Text("Home")),
				Column(
					[
					Text(page.route),

					ElevatedButton(
						"ini home",
						on_click=page.go("/store/""ewqeq")
						)
					],
					alignment="center"

					)

				]
			)
			)
		if page.route == "/store":
			page.views.append(
			View(
				"/store/:id",
				[
				AppBar(title=Text("store")),
				Column(
					[
		Text(page.route),
					
					ElevatedButton(
						"ini store",
						on_click=view_pop
						)
					],
					alignment="center"

					)

				]
			)
			)
```
i want to route **/store/youid**
and display parameter in /store Screen
is there way for solution

and one again
**how to redirect automaticaly , if user is not authentication**
thx

**Comment by FeodorFitsner** [comment]
So, right you do `page.go(f"/store/{store_id}")` to navigate to a store page.
Then you have to parse `store_id` from a route in `route_change` handler, using regexp or string split. Right now there is no automatic pattern matching like `/store/:id`.

**Comment by bobwatcherx** [comment]
outside the theme .  can i create a function inside the widget example

 **TextField(
visible = if is_auth == true ? 'true':'false'
)**


 property or is there a way to write it ?thx

**Reply by FeodorFitsner** [reply]
I don't think such a reactive approach is available now, unless you explicitly re-create `TextField` control.

**Reply by ndonkoHenri** [reply]
I don't know if I perfectly understood your question, but just want to mention that the below is possible. 🙂

```python
TextField(
   visible = True if is_auth == True else False
)
```

Hope it helps.
